extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -304,6 +352,24 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
     });
   }
 
+  /**
+   * onclick event that fires when buy now is clicked,
+   * sends an analytics event containing product id and the conversion url.
+   * @param {!ShoppingConfigDataDef} activeProductData
+   * @return {boolean}
+   * @private
+   */
+  onClickBuyNow_(activeProductData) {
+    this.variableService_.onVariableUpdate(
+      AnalyticsVariable.STORY_SHOPPING_PRODUCT_ID,
+      activeProductData.productId
+    );
```

Comment:
Is this variable already set correctly since we set it when the template updates?
Do we need to set it again?

Reply:
Good point, we could eliminate this redundancy...